cscs-checks/microbenchmarks/gpu/memory_bandwidth/src/Xdevice/common/smi.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef __DEFINED_COMMON_SMI__
+#define __DEFINED_COMMON_SMI__
+
+#include <iostream>
+
+class Smi
+{
+private:
+  /*
+   * Indicator that shows if the Smi class has any active instances.
+   *
+   * This is set to 1 if the activeSmiInstances is greater that 0.
+   *
+   */
+  static int smiIsActive;
+
+  /*
+   * Active instance counter.
+   *
+   * This counter is incrementer/decremented in the class constructor/destructor.
+   *
+   */
+  static int activeSmiInstances;
+
+  /*
+   * Number of present devices in the current node.
+   *
+   * This gets set during the class construction.
+   *
+   */
+  unsigned int numberOfDevices;
+
+public:
+  Smi();
+  ~Smi();
+
+  /*
+   * Gets the number of devices.
+   *
+   * Parameters:
+   *   + n: address of the integer where to return the number of devices.
+   */
+  void getNumberOfDevices(int * n);
+
+  /*
+   * Test wheter the requested GPU id is within the range of available IDs in the curent node.
+   *
+   * This function can exit the program if the requested value is outside the valid range.
+   *
+   * Parameters:
+   *   + id: Device ID.
+   */
+  void checkGpuIdIsSensible(int id);
+
+  /*
+   * Pin the host thread to a CPU with affinity to a given device.
+   *
+   * Parameters:
+   *   + id: Device ID.
+   */
+  void setCpuAffinity(int id);
+
+  /*
+   * Get the current temperature of a given device.
+   *
+   * Parameters:
+   *   + id: Device ID.
+   */
+  float getGpuTemp(int id);
+
+  /*
+   * Get the total memory size present in a device.
+   *
+   * Parameters:
+   *   + id: Device id.
+   *   + mSize: Size (in bytes) of the total memory.
+   *
+   */
+  void getDeviceMemorySize(int id, size_t * mSize);
```

Comment:
I would probably mark these as `const`. Btw, why don't you return the size and pass it as a pointer instead?

Reply:
All these functions are void just to be consistent with both ROCm-SMI and NVML implementations. The functions in these libraries return a status, not the requested value. Actually, I should probably also change the getGpuTemp to be consistent with the others.